src/main/java/com/booleanuk/Poker.java

```diff
@@ -28,4 +99,70 @@ public String extensionThreeMethods() {
         // You will also need to add the relevant tests to Extension3Test.java
         return "Replace this method with your own methods and tests";
     }
+
+    public String getCardOfPair(String[] hand){
+        String resCard = "";
+        String firstCard = hand[0];
+        String previousCard = "";
+        for (String card: hand){
+            if(isPair(new String[]{previousCard, card}) || isPair(new String[]{firstCard, card})){
+                resCard = card;
+            }
+            previousCard = card;
+        }
+        return resCard;
+    }
+
+    public int amountOfIdenticalCards(String[] hand){
+        int amount = 0;
+        String firstCard = hand[0];
+        String previousCard = "";
+        for (String card: hand){
+            if(isPair(new String[]{previousCard, card}) || isPair(new String[]{firstCard, card})){
+                amount +=1;
+            }
+            previousCard = card;
+        }
+        return amount;
+    }
+
+    public boolean isPair(String[] hand){
+        return hand[0].equalsIgnoreCase(hand[1]);
+    }
+
+    public String getHigestCard(ArrayList<String> listOfCards){
+
+        int maxValue = 0;
+        String highestCard = "";
+
+        for (String card : listOfCards){
+            int currentValue = this.cardValueMap.get(card);
+            if(currentValue > maxValue){
+                maxValue = currentValue;
+                highestCard = card;
+            }
+        }
+        return highestCard;
+    }
+
+    public HashMap<String, Integer> createCardValueMap(){
+
+        HashMap<String, Integer> cardValueMap = new HashMap<>();
+
+        cardValueMap.put("2", 2);
+        cardValueMap.put("3", 3);
+        cardValueMap.put("4", 4);
+        cardValueMap.put("5", 5);
+        cardValueMap.put("6", 6);
+        cardValueMap.put("7", 7);
+        cardValueMap.put("8", 8);
+        cardValueMap.put("9", 9);
+        cardValueMap.put("10", 10);
+        cardValueMap.put("J", 11);
+        cardValueMap.put("Q", 12);
+        cardValueMap.put("K", 13);
+        cardValueMap.put("A", 14);
+
+        return cardValueMap;
+    }
 }
```

Comment:
Your approach, especially using a HashMap for card values, was smart and makes your code easy to follow. I like how you handled both pairs and three-of-a-kinds, especially with three-card hands – it shows you've thought about the different scenarios. Keep up the good work.